cassandra/cluster.py

```diff
@@ -2654,6 +2675,12 @@ def has_more_pages(self):
         """
         return self._paging_state is not None
 
+    @property
+    def custom_payload(self):
+        if not self._event.is_set():
+            raise Exception("custom_payload cannot be retrieved before ResponseFuture is finalized")
```

Comment:
Good question. If we wait for the future, then there has to be a timeout, which doesn't feel right for this property. This might be a good candidate for change once I get timers introduced and allow true request timeouts.

In the mean time, do you think it should just return what's there, regardless of whether it's been set yet (instead of throwing)? lmkwyt
Regardless, I should add a docstring that describes the semantics.